dist/ethereum.js

```diff
@@ -2580,23 +2583,31 @@ var requestManager = function() {
     var send = function (data, callback) {
         /*jshint maxcomplexity: 8 */
 
-        // FORMAT BASED ON ONE FORMATTER function
-        if(typeof data.inputFormatter === 'function') {
-            data.params = Array.prototype.map.call(data.params, function(item, index){
-                // format everything besides the defaultblock, which is already formated
-                return (!data.addDefaultblock || index+1 < data.addDefaultblock) ? data.inputFormatter(item) : item;
-            });
+        if(!utils.isArray(data))
+            data = [data];
 
-        // FORMAT BASED ON the input FORMATTER ARRAY
-        } else if(data.inputFormatter instanceof Array) {
-            data.params = Array.prototype.map.call(data.inputFormatter, function(formatter, index){
-                // format everything besides the defaultblock, which is already formated
-                return (!data.addDefaultblock || index+1 < data.addDefaultblock) ? formatter(data.params[index]) : data.params[index];
-            });
-        }
+        var payload = [];
 
 
-        var payload = jsonrpc.toPayload(data.method, data.params);
+        data.forEach(function(item){
+            // FORMAT BASED ON ONE FORMATTER function
+            if(typeof item.inputFormatter === 'function') {
+                item.params = Array.prototype.map.call(item.params, function(param, index){
+                    // format everything besides the defaultblock, which is already formated
+                    return (!item.addDefaultblock || index+1 < item.addDefaultblock) ? item.inputFormatter(param) : param;
+                });
+
+            // FORMAT BASED ON the input FORMATTER ARRAY
+            } else if(item.inputFormatter instanceof Array) {
+                item.params = Array.prototype.map.call(item.inputFormatter, function(formatter, index){
+                    // format everything besides the defaultblock, which is already formated
+                    return (!item.addDefaultblock || index+1 < item.addDefaultblock) ? formatter(item.params[index]) : item.params[index];
+                });
+            }
+
+            payload.push(jsonrpc.toPayload(item.method, item.params));
```

Comment:
1. agree i will take a look
2. true, but better we use undescore anyway ad we introduce more and more of thos funcitons.
3. i thinks thats actually clever, as you allow the specify one inputformatter for all parameters, or one for each single parameter. This way the developer of the method can choose what he wants. And its actually a better way of using the types here, as different types (array or funciton) mean different behaviour.
4. You would like to move the formating into 30 different places spread all over ethreuem.js?
Its called send, but actually its called Requestmanager.send(), means its a module and what the requestmanager need to do before he sends it is not important for other modules to know, so i think it belongs here. Though i agree we could split up the functionality more, inside requestmanager.